File: tests/functional/Bots/Python/Skills/CodeFirst/WaterfallSkillBot/dialogs/proactive/continuation_parameters.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from botbuilder.schema import ConversationReference


class ContinuationParameters:
    def __init__(
        self,
        claims_identity: str,
        oauth_scope: str,
        conversation_reference: ConversationReference,
    ):
        self.claims_identity = claims_identity
        self.oauth_scope = oauth_scope
        self.conversation_reference = conversation_reference

File: tests/functional/Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/proactive/continuationParameters.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

/**
 * Stores the information needed to resume a conversation when a proactive message arrives.
 */
class ContinuationParameters {
  /**
   * @param {string} claimsIdentity
   * @param {string} oAuthScope
   * @param {import('botbuilder').ConversationReference} conversationReference
   */
  constructor (claimsIdentity, oAuthScope, conversationReference) {
    this.claimsIdentity = claimsIdentity;
    this.oAuthScope = oAuthScope;
    this.conversationReference = conversationReference;
  }
}

module.exports.ContinuationParameters = ContinuationParameters;
